app/src/main/java/org/wikipedia/WikipediaApp.kt

```diff
@@ -251,16 +251,20 @@ class WikipediaApp : Application() {
     }
 
     fun resetAfterLogOut() {
-        AccountUtil.removeAccount()
-        Prefs.isPushNotificationTokenSubscribed = false
-        Prefs.pushNotificationTokenOld = ""
-        Prefs.tempAccountWelcomeShown = false
-        Prefs.tempAccountCreateDay = 0L
-        Prefs.tempAccountDialogShown = false
-        SharedPreferenceCookieManager.instance.clearAllCookies()
-        AppDatabase.instance.notificationDao().deleteAll()
-        FlowEventBus.post(LoggedOutEvent())
-        L.d("Logout complete.")
+        MainScope().launch(CoroutineExceptionHandler { _, t ->
+            L.e(t)
+        }) {
```

Comment:
This function must not launch a coroutine. It needs to be run synchronously, so that the logic of `CsrfTokenClient` can work correctly.